src/app/assessment/assessment.service.ts

```diff
@@ -366,31 +372,33 @@ export class AssessmentService {
     return answer;
   }
 
-  saveAnswers(assessment: AssessmentSubmitBody, answers: object, action: string, submissionId?: number) {
-    let postData;
-    switch (action) {
-      case 'assessment':
-        postData = {
-          Assessment: assessment,
-          AssessmentSubmissionAnswer: answers
-        };
-        if (submissionId) {
-          postData.AssessmentSubmission = {
-            id: submissionId
-          };
-        }
-        return this.request.post(api.post.submissions, postData);
-
-      case 'review':
-        postData = {
-          Assessment: assessment,
-          AssessmentReviewAnswer: answers
-        };
-        return this.request.post(api.post.reviews, postData);
+  saveAnswers(assessment: AssessmentSubmitParams, answers: Answer[], action: string) {
+    if (!['assessment', 'review'].includes(action)) {
+      return of(false);
     }
-    return of({
-      success: false
+    let params = `assessmentId:${assessment.id},inProgress:${assessment.inProgress},answers:${this._answersFormatter(answers)}`;
+    ['submissionId', 'contextId', 'reviewId', 'unlock'].forEach(key => {
+      if (assessment[key]) {
+        params += `,${key}:${assessment[key]}`;
+      }
     });
+    return this.request.postGraphQL(this.utils.graphQLQueryStringFormatter(`"
+      mutation {
+        ` + (action === 'assessment' ? `submitAssessment` : `submitReview`) + `(` + params + `)
+      }
+    "`));
+  }
+
+  // re-format answers array so that it can be passed to GraphQL
+  private _answersFormatter(answers: Answer[]) {
```

Comment:
@shawnm0705 ignore this comment, I'm commenting to leave a note for myself for reference in the future. (Cuz i believe graphql themselves should handle this instead letting consumer handling this, although it's not impossible, but i dont feel safe handling this by ourselves)